### PR TITLE
feat: vertically center pfe-tab content

### PR DESCRIPTION
**Pull request by bennypowers**
### Related issues

- (#1086) Vertically align tab text

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-1737--patternfly-elements.netlify.app/components/tabs/) 
Screen Shot 2021-08-01 at 0 21 10

### What has changed and why

- `pfe-tab.scss` changed to make tab content flex and take up 100% of containter height.

### Testing instructions

- [x] *Test case 1*
  1. Go to [`/components/tabs`](https://deploy-preview-1737--patternfly-elements.netlify.app/components/tabs/) on the demo site
  2. Change the outerHTML of a horizontal tabs element to something with lots of text, preferably with a nautical or showtunes theme, like this:
  
```html
<pfe-tabs tab-align="center" aria-orientation="horizontal" variant="wind" role="tablist" pfelement="" class="PFElement" type="combo" has_tab="" has_panel="" selected-index="0" on="light">
  <pfe-tab role="tab" slot="tab" aria-selected="true" tabindex="0" pfelement="" class="PFElement" type="content" id="pfe-en1kv722o" variant="wind" aria-controls="pfe-lz2trfbry" on="light">
    <h1>Tab 1</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-lz2trfbry" variant="wind" aria-labelledby="pfe-en1kv722o" on="light">
    <h2>Content 1</h2>
    <p>Tab 1 panel content.</p>
  </pfe-tab-panel>
  <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1>Tab 2</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
  
    <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1>I am the very model</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
    <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1>Of a modern Major General</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
  
    <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1>I've information vegetable</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
  
    <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1>animal and mineral</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
  
    <pfe-tab role="tab" slot="tab" aria-selected="false" tabindex="-1" pfelement="" class="PFElement" type="content" id="pfe-jsliputv6" variant="wind" aria-controls="pfe-kcytgau8t" on="light">
    <h1> I know the kings of England and I quote the fights historical, From Marathon to Waterloo in order categorical.</h1>
  </pfe-tab>
  <pfe-tab-panel role="tabpanel" slot="panel" tabindex="0" pfelement="" class="PFElement" type="container" id="pfe-kcytgau8t" variant="wind" aria-labelledby="pfe-jsliputv6" on="light" hidden="">
    <h2>Content 2</h2>
    <p>Tab 2 panel content.</p>
  </pfe-tab-panel>
</pfe-tabs> 
```

#### Browser requirements

Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Internet Explorer 11 (should be useable, not pixel perfect)
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [ ] Firefox 78 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Android mobile device (such as the Galaxy S9)
- [x] Apple mobile device (such as the iPhone X)
- [x] Apple tablet device (such as the iPhone Pro)

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Tests have been updated to cover these changes.
- [x] Browser testing passed.
- [x] Changelog updated.
- [x] Documentation (README.md, WHY.md, etc.) updated or added.
- [ ] Link to the demo recording: []()
- [x] Approved by designer.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

**Comment by bennypowers** [comment]
maintainers: Not sure how much of the documentation/changelog needs updating here, and if needed, how to go about updating the relevant changelog. ditto for tests, although `npm t` passed locally. guidance is appreciated 🙏 .

**Comment by castastrophe** [comment]
> preferably with a nautical or showtunes theme, like this

😆 ❤️ it

File: elements/pfe-tabs/src/pfe-tab.scss
Comment (castastrophe) [inline elements/pfe-tabs/src/pfe-tab.scss:55]: This is solid btw.  I put really janky content in here to try and break it and it stays nicely centered. :)

**Comment by castastrophe** [comment]
@bennypowers You can update the CHANGELOG-1.x.md file in the root of the project by adding a placeholder (we won't have the commit hash until after merging) - this will be automated eventually I'm sure, we just never had time while I was there :)

- [](https://github.com/patternfly/patternfly-elements/commit/) feat: vertically center pfe-tab content

Add that under the # 1.11.0 (2021) heading

**Comment by bennypowers** [comment]
> @bennypowers You can update the CHANGELOG-1.x.md file in the root of the project by adding a placeholder (we won't have the commit hash until after merging) - this will be automated eventually I'm sure, we just never had time while I was there :)
> 
>     * feat: vertically center pfe-tab content
> 
> 
> Add that under the # 1.11.0 (2021) heading

done, thanks

**Comment by castastrophe** [comment]
@bennypowers Re your question about tests, I haven't spent much energy writing AT for styles on the components so I think you can skip it.

**Comment by bennypowers** [comment]
Could write some visual diff tests with  playwright but that's maybe for another time

**Comment by heyMP** [comment]
Thank you @bennypowers!!! Sorry for the delay!  I'll browser check this and get it merged.  Thanks for the contribution 🎉

**Comment by heyMP** [comment]
@bennypowers, we should probably update the `variant="earth" vertical` variant to `display: inline-flex` as well.  

Screen Shot 2021-08-06 at 10 36 39 AM

This one is tricky because even though it's a "vertical" variant it actually transitions to "horizontal" layout on mobile.

Here is the selector: https://github.com/patternfly/patternfly-elements/blob/1f0b6693e5c7c47bdaba4b5afe1013986953844f/elements/pfe-tabs/src/pfe-tab.scss#L92

**Comment by bennypowers** [comment]
sounds good. I'll get to this sunday or monday. have a nice weekend

**Comment by heyMP** [comment]
You rock Benny.  Thank you! 😀

**Comment by bennypowers** [comment]
heads up i force pushed and squashed. to bring `vertical variant="earth"` in, I just removed that entire rule.

**Review by heyMP — APPROVED** [review]
Looks goood to me :)

**Pull request merged**
